### PR TITLE
MatekH743 board_id, usb vid/pid changes, and MPU6000 delay before transfer - not after

### PR DESCRIPTION
**Pull request by vvladic**
Please use [PX4 Discuss](http://discuss.px4.io/) or [Slack](http://slack.px4.io/) to align on pull requests if necessary. You can then open draft pull requests to get early feedback.

**Describe problem solved by this pull request**
Board_id not compatible with ardupilot, uninitialized gpio outputs, delay before transfer not after in MPU6000

**Describe your solution**
Corrected board id, initialized outputs, moved delay before spi transfer

**Describe possible alternatives**
A clear and concise description of alternative solutions or features you've considered.

**Test data / coverage**
Bench tested: bootloader OK, pins enabled permanently for now, MPU6000 delay moved before spi transfer

### CODE REVIEW

File: boards/matek/h743-slim/nuttx-config/bootloader/defconfig [hunk 1]
```diff
@@ -29,12 +29,12 @@ CONFIG_BOARD_LOOPSPERMSEC=22114
...
-CONFIG_CDCACM_PRODUCTID=0x004b
-CONFIG_CDCACM_PRODUCTSTR="Matek H743-slim"
+CONFIG_CDCACM_PRODUCTID=0x1003
+CONFIG_CDCACM_PRODUCTSTR="MatekH743"
 CONFIG_CDCACM_RXBUFSIZE=600
 CONFIG_CDCACM_TXBUFSIZE=12000
-CONFIG_CDCACM_VENDORID=0x3162
-CONFIG_CDCACM_VENDORSTR="Matek"
+CONFIG_CDCACM_VENDORID=0x1209
+CONFIG_CDCACM_VENDORSTR="PX4"
```
Comment (dagar) [inline boards/matek/h743-slim/nuttx-config/bootloader/defconfig:37]: The vendor string as Matek seems fine.

Comment (vvladic) [reply boards/matek/h743-slim/nuttx-config/bootloader/defconfig]: ok changed it back

**Comment by taileron** [comment]
@vvladic how about remove the suffix "slim" from your target which, by the way, also runs perfectly on the mini fc. Having an unified target that should run on all variants of this fc (like AP does). By adding appropriate modules, mc, fw and vtol would be feasible with the code. #18730 Only one folder would have to be maintained ...

**Comment by vvladic** [comment]
> @vvladic how about remove the suffix "slim" from your target which, by the way, also runs perfectly on the mini fc. Having an unified target that should run on all variants of this fc (like AP does). By adding appropriate modules, mc, fw and vtol would be feasible with the code. #18730 Only one folder would have to be maintained ...

We can do this in another pull request perhaps, after we merge this one. Would that be ok?

**Comment by vvladic** [comment]
@dagar Can we merge this? Is there anything else that you wish to be done before merging?

**Comment by dagar** [comment]
> @dagar Can we merge this? Is there anything else that you wish to be done before merging?

Are you sure the flash based parameters are working on H7? Last I heard there was an issue due to the ECC.

**Comment by dagar** [comment]
Maybe leave the flash based parameter changes commented and we'll merge the rest right now?

**Comment by vvladic** [comment]
> Maybe leave the flash based parameter changes commented and we'll merge the rest right now?

It's already disabled in board_config.h
https://github.com/PX4/PX4-Autopilot/blob/5b07398b3e02ccd3a077ec57e5a44fa6ea7d5fe6/boards/matek/h743-slim/src/board_config.h#L52-L57

Would you like me to comment it out of src/init.c as well?

**Review by dagar — APPROVED** [review]

**Comment by dagar** [comment]
Thanks @vvladic

**Comment by taileron** [comment]
> > @vvladic how about remove the suffix "slim" from your target which, by the way, also runs perfectly on the mini fc. Having an unified target that should run on all variants of this fc (like AP does). By adding appropriate modules, mc, fw and vtol would be feasible with the code. #18730 Only one folder would have to be maintained ...
> 
> We can do this in another pull request perhaps, after we merge this one. Would that be ok?

no problem, but this board now also uses IMU ICM42605 instead of ICM20602 in the current version

**Pull request merged**